src/Build/Evaluation/ItemSpec.cs

```diff
@@ -244,22 +257,52 @@ internal abstract class ItemFragment
         /// </summary>
         protected string ProjectDirectory { get; }
 
-        private readonly Lazy<FileSpecMatcherTester> _fileMatcher;
-        private readonly Lazy<IMSBuildGlob> _msbuildGlob;
-        protected virtual IMSBuildGlob MsBuildGlob => _msbuildGlob.Value;
+        private bool _fileMatcherInitialized;
+        private FileSpecMatcherTester _fileMatcher;
+
+        // not a Lazy to reduce memory
+        private FileSpecMatcherTester FileMatcher
+        {
+            get
+            {
+                if (_fileMatcherInitialized)
+                {
+                    return _fileMatcher;
+                }
+
+                _fileMatcher = CreateFileSpecMatcher();
+                _fileMatcherInitialized = true;
+
+                return _fileMatcher;
+            }
+        }
+
+        private IMSBuildGlob _msbuildGlob;
+
+        // not a Lazy to reduce memory
+        protected virtual IMSBuildGlob MsBuildGlob
+        {
+            get
+            {
+                if (_msbuildGlob == null)
+                {
+                    _msbuildGlob = CreateMsBuildGlob();
```

Comment:
You changed the creation semantics here - this could be created multiple times - is that a problem?